src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisResultService.java

```diff
@@ -0,0 +1,207 @@
+            securityAnalysisResult.setStatus(status);
+            securityAnalysisResultRepository.save(securityAnalysisResult);
+        });
+    }
+
+    @Transactional
+    public void delete(UUID resultUuid) {
+        Objects.requireNonNull(resultUuid);
+        securityAnalysisResultRepository.deleteById(resultUuid);
+    }
+
+    @Transactional
+    public void deleteAll() {
+        securityAnalysisResultRepository.deleteAll();
+    }
+
+    @Transactional(readOnly = true)
+    public SecurityAnalysisStatus findStatus(UUID resultUuid) {
+        Objects.requireNonNull(resultUuid);
+        Optional<SecurityAnalysisResultEntity> securityAnalysisResult = securityAnalysisResultRepository.findById(resultUuid);
+        if (securityAnalysisResult.isEmpty()) {
+            return null;
+        }
+
+        return securityAnalysisResult.get().getStatus();
+    }
+
+    private static LimitViolation fromEntity(AbstractLimitViolationEntity limitViolationEntity) {
+        String subjectId = limitViolationEntity.getConstraint() != null
+            ? limitViolationEntity.getConstraint().getSubjectId()
+            : null;
+
+        return new LimitViolation(subjectId, limitViolationEntity.getLimitType(), limitViolationEntity.getLimitName(), limitViolationEntity.getAcceptableDuration(),
+            limitViolationEntity.getLimit(), limitViolationEntity.getLimitReduction(), limitViolationEntity.getValue(), limitViolationEntity.getSide());
+    }
+
+    private static SecurityAnalysisResultEntity toEntity(UUID resultUuid, SecurityAnalysisResult securityAnalysisResult, SecurityAnalysisStatus securityAnalysisStatus) {
+        Map<String, ConstraintEntity> constraintsBySubjectId = getUniqueConstraintsFromResult(securityAnalysisResult)
+            .stream().collect(Collectors.toMap(
+                ConstraintEntity::getSubjectId,
+                constraint -> constraint)
+            );
+
+        List<ContingencyEntity> contingencies = securityAnalysisResult.getPostContingencyResults().stream()
+            .map(postContingencyResult -> toEntity(postContingencyResult, constraintsBySubjectId)).collect(Collectors.toList());
+
+        List<PreContingencyLimitViolationEntity> preContingencyLimitViolations = toEntity(securityAnalysisResult.getPreContingencyResult(), constraintsBySubjectId);
+
+        return new SecurityAnalysisResultEntity(resultUuid, securityAnalysisStatus, securityAnalysisResult.getPreContingencyResult().getStatus().name(), contingencies, preContingencyLimitViolations);
+    }
+
+    private static List<ConstraintEntity> getUniqueConstraintsFromResult(SecurityAnalysisResult securityAnalysisResult) {
+        return Stream.concat(
+                securityAnalysisResult.getPostContingencyResults().stream().flatMap(pcr -> pcr.getLimitViolationsResult().getLimitViolations().stream()),
+                securityAnalysisResult.getPreContingencyResult().getLimitViolationsResult().getLimitViolations().stream())
```

Comment:
same remark as above : Are you sure you should take into account the preContingencyLimitViolations collection here ?
The constraints are only needed when we access the NmK results, no ?

Reply:
same response as above, we should discuss it